allauth/account/internal/flows/login_by_code.py

```diff
@@ -28,7 +28,8 @@ def request_login_code(request: HttpRequest, email: str) -> None:
         "failed_attempts": 0,
     }
     if not users:
-        send_unknown_account_mail(request, email)
+        if app_settings.EMAIL_UNKNOWN_ACCOUNTS:
+            send_unknown_account_mail(request, email)
```

Comment:
Instead of duplicating this `if`, I think it would be better to fold this into the `send_unknown_account_mail` implementation itself.

Reply:
Sounds good. Additional commit added to do this.